server/app/views/questiontypes/MapQuestionFragment.html

```diff
@@ -5,17 +5,33 @@
   th:fragment="map(question, questionRendererParams)"
   th:with="mapQuestion=${question.createMapQuestion()},
              questionId=${'id-' + #strings.randomAlphanumeric(8)},
-             firstPathWithError=${mapQuestion.getFirstPathWithError()},
+              mapId=${'cf-map-' + #strings.randomAlphanumeric(8)},
+              firstPathWithError=${mapQuestion.getFirstPathWithError()},
              hasGroupErrors=${mapQuestion.getValidationErrors().get(question.getContextualizedPath())},"
   th:id="${questionId}"
   class="cf-question-map"
   th:classappend="${hasGroupErrors ? 'cf-question-field-with-error' : ''}"
   data-testid="questionRoot"
 >
   <!-- TODO(#11003): Build out applicant view of map question -->
-  This is a map question!
+  <div th:id="${mapId}" class="square-mobile"></div>
   <div
+    th:data-mapid="|${mapId}|"
+    class="hidden"
     th:if="${questionRendererParams.geoJson().isPresent()}"
-    th:text="${questionRendererParams.geoJson().get()}"
+    th:value="${questionRendererParams.geoJson().get()}"
   ></div>
+  <script th:inline="javascript" th:nonce="${cspNonce}">
+    window.app = window.app || {}
+    window.app.data = window.app.data || {}
+    window.app.data.maps = window.app.data.maps || {}
+
+    id = /*[[${mapId}]]*/ 'id'
+    if (
+      /*[[${questionRendererParams.geoJson().isPresent()}]]*/ mapDataIsPresent
+    ) {
+      window.app.data.maps[id] =
+        /*[[${questionRendererParams.geoJson().get()}]]*/ {}
+    }
+  </script>
```

Comment:
```suggestion
  <script
    th:if="${questionRendererParams.geoJson().isPresent()}"
    th:inline="javascript" th:nonce="${cspNonce}">
    window.app = window.app || {};
    window.app.data = window.app.data || {};
    window.app.data.maps = window.app.data.maps || {};
    window.app.data.maps[/*[[${mapId}]]*/ 'id'] = JSON.parse(/*[[${questionRendererParams.geoJson().get()}]]*/ {});
  </script>
```

You should be able to do this. Only render the script tag if the value exists. That simplifies the javascript a lot. Parse the JSON in here as in this example or keep doing it in the ts method.